src/3d/qgsannotationlayerchunkloader_p.cpp

```diff
@@ -0,0 +1,396 @@
+  itemIds.unite( layer->mNonIndexedItems );
+
+  mItemsToRender.reserve( itemIds.size() );
+  std::transform( itemIds.begin(), itemIds.end(), std::back_inserter( mItemsToRender ), [layer]( const QString &id ) -> std::unique_ptr< QgsAnnotationItem > {
+    return std::unique_ptr< QgsAnnotationItem >( layer->item( id )->clone() );
+  } );
+
+  //
+  // this will be run in a background thread
+  //
+  mFutureWatcher = new QFutureWatcher<void>( this );
+  connect( mFutureWatcher, &QFutureWatcher<void>::finished, this, &QgsChunkQueueJob::finished );
+
+  const QFuture<void> future = QtConcurrent::run( [this, rect, layerToMapTransform, zOffset, altitudeClamping, showCallouts] {
+    const QgsEventTracing::ScopedEvent e( QStringLiteral( "3D" ), QStringLiteral( "Annotation layer chunk load" ) );
+
+    std::vector< Billboard > billboards;
+    billboards.reserve( mItemsToRender.size() );
+    QVector< QImage > textures;
+    textures.reserve( mItemsToRender.size() );
+
+    for ( const std::unique_ptr< QgsAnnotationItem > &item : std::as_const( mItemsToRender ) )
+    {
+      if ( mCanceled )
+        break;
+
+      QgsAnnotationItem *annotation = item.get();
+
+      if ( !annotation->enabled() )
+        continue;
+
+      if ( QgsAnnotationMarkerItem *marker = dynamic_cast< QgsAnnotationMarkerItem * >( annotation ) )
+      {
+        if ( marker->symbol() )
+        {
+          QgsPointXY p = marker->geometry();
+          try
+          {
+            const QgsPointXY mapPoint = layerToMapTransform.transform( p );
+            if ( !rect.contains( mapPoint ) )
+              continue;
+
+            double z = 0;
+            const float terrainZ = ( altitudeClamping == Qgis::AltitudeClamping::Absolute && !showCallouts ) ? 0 : mRenderContext.terrainRenderingEnabled() && mRenderContext.terrainGenerator() ? static_cast<float>( mRenderContext.terrainGenerator()->heightAt( mapPoint.x(), mapPoint.y(), mRenderContext ) * mRenderContext.terrainSettings()->verticalScale() )
+                                                                                                                                                                                                 : 0.f;
+
+            switch ( altitudeClamping )
+            {
+              case Qgis::AltitudeClamping::Absolute:
+                z = zOffset;
+                break;
+              case Qgis::AltitudeClamping::Terrain:
+                z = terrainZ;
+                break;
+              case Qgis::AltitudeClamping::Relative:
+                z = terrainZ + zOffset;
+                break;
+            }
+
+            Billboard billboard;
+            billboard.position = ( QgsVector3D( mapPoint.x(), mapPoint.y(), z ) - mChunkOrigin ).toVector3D();
+            billboard.textureId = textures.size();
+            textures.append( QgsPoint3DBillboardMaterial::renderSymbolToImage( marker->symbol(), mRenderContext ) );
```

Comment:
I was looking at upstream, and saw that this is deprecated 
https://doc.qt.io/qt-6/qt3drender-module-obsolete.html)

I suppose this is a broader issue than the scope of this PR?

I was actually wondering if there is some caching and where it would/need to happen.

Reply:
OK, I wasn't aware of this deprecation.